arelle/plugin/validate/DBA/rules/__init__.py

```diff
@@ -58,20 +58,32 @@ def errorOnRequiredFact(
         factQn: QName,
         code: str,
         message: str,
+        level: str = 'error',
+        modelObject: list[ModelFact] | None = None,
 ) -> Iterable[Validation]:
     """
     Yields an error if a fact with the given QName is not tagged with a non-nil value.
     :return: Yields validation errors.
     """
+    if modelObject is None:
+        modelObject = []
     facts: set[ModelFact] = modelXbrl.factsByQname.get(factQn, set())
     if facts:
         for fact in facts:
             if not fact.isNil:
                 return
-    yield Validation.error(
-        codes=code,
-        msg=message,
-)
+    if level == 'error':
```

Comment:
I suggest at least replacing `level: str = 'error'` with `isError: bool = True` so we aren't depending on "magic" strings. But a step further, I might suggest just repurposing this to simply `isFactMissing` and let the caller decide how it wants to call `Validation.error` or `Validation.warning`.